ingestion/src/metadata/ingestion/source/objectstore/s3/metadata.py

```diff
@@ -0,0 +1,287 @@
+from metadata.utils.s3_utils import prefix_exits
+
+logger = ingestion_logger()
+
+OPENMETADATA_TEMPLATE_FILE_NAME = "openmetadata.json"
+
+
+class S3Metric(Enum):
+    NUMBER_OF_OBJECTS = "NumberOfObjects"
+    BUCKET_SIZE_BYTES = "BucketSizeBytes"
+
+
+class S3BucketResponse(BaseModel):
+    """
+    Class modelling a response received from s3_client.list_buckets operation
+    """
+
+    class Config:
+        extra = Extra.forbid
+
+    Name: str = Field(..., description="Bucket name", title="Bucket Name")
+    CreationDate: datetime = Field(
+        ...,
+        description="Timestamp of Bucket creation in ISO format",
+        title="Creation Timestamp",
+    )
+
+
+class S3ContainerDetails(BaseModel):
+    class Config:
+        extra = Extra.forbid
+
+    name: str = Field(..., description="Bucket name", title="Bucket Name")
+    numberOfObjects: float = Field(..., description="Total nr. of objects", title="Nr. of objects")
+    size: float = Field(..., description="Total size in bytes of all objects", title="Total size(bytes) of objects")
+    fileFormats: Optional[List[container.FileFormat]] = Field(..., description="File formats", title="File formats")
+    dataModel: Optional[ContainerDataModel] = Field(..., description="Data Model for the container", title="Data Model")
+    CreationDate: str = Field(
+        ...,
+        description="Timestamp of Bucket creation in ISO format",
+        title="Creation Timestamp",
+    )
+
+
+class S3Source(ObjectStoreServiceSource):
+    """
+    Source implementation to ingest S3 buckets data.
+    """
+
+    def __init__(self, config: WorkflowSource, metadata_config: OpenMetadataConnection):
+        super().__init__(config, metadata_config)
+        self.s3_client = self.connection.s3_client
+        self.cloudwatch_client = self.connection.cloudwatch_client
+
+    @classmethod
+    def create(cls, config_dict, metadata_config: OpenMetadataConnection):
+        config: WorkflowSource = WorkflowSource.parse_obj(config_dict)
+        connection: S3StoreConnection = config.serviceConnection.__root__.config
+        if not isinstance(connection, S3StoreConnection):
+            raise InvalidSourceException(
+                f"Expected S3StoreConnection, but got {connection}"
+            )
+        return cls(config, metadata_config)
+
+    def get_containers(self) -> Iterable[S3ContainerDetails]:
+        """
+        List and filter containers
+        """
+        buckets = []
+        try:
+            buckets = self.s3_client.list_buckets()['Buckets']
+        except Exception as err:
+            logger.debug(traceback.format_exc())
+            logger.error(f"Failed to fetch buckets list - {err}")
+
+        # No pagination required, as there is a hard 1000 limit on nr of buckets per aws account
+        for bucket in buckets:
+            bucket_response: S3BucketResponse = S3BucketResponse.parse_obj(bucket)
+            bucket_name = bucket_response.Name
```

Comment:
not sure we need this assignation since we have the model already prepared

Reply:
Em, well this was used in 3 places and I just thought **bucket_name** is shorter and clearer instead of using **bucket_reponse.Name** everywhere.